pkg/controller/controller_test.go

```diff
@@ -643,3 +644,145 @@ func TestDeleteVolumeGroup(t *testing.T) {
 		t.Errorf("Failed to delete volume group: %v\n", err)
 	}
 }
+
+func NewFakeFileShareController() fileshare.Controller {
+	return &fakeFileShareController{}
+}
+
+type fakeFileShareController struct {}
+
+func (fakeFileShareController) SetDock(dockInfo *model.DockSpec) { return }
+
+func (fakeFileShareController) CreateFileShare(opt *pb.CreateFileShareOpts) (*model.FileShareSpec, error) {
+	return &SampleFileShares[0], nil
+}
+
+func (fakeFileShareController) CreateFileShareAcl(opt *pb.CreateFileShareAclOpts) (*model.FileShareAclSpec, error) {
+	return &SampleFileSharesAcl[0], nil
+}
+
+func (fakeFileShareController) DeleteFileShareAcl(opt *pb.DeleteFileShareAclOpts) error {return nil}
+
+func (fakeFileShareController) DeleteFileShare(opt *pb.DeleteFileShareOpts) error {return nil}
+
+func (fakeFileShareController) CreateFileShareSnapshot(opt *pb.CreateFileShareSnapshotOpts) (*model.FileShareSnapshotSpec, error) {
+	return &SampleFileShareSnapshots[0], nil
+}
+
+func (fakeFileShareController) DeleteFileShareSnapshot(opts *pb.DeleteFileShareSnapshotOpts) error {return nil}
+
+func TestCreateFileShare(t *testing.T) {
+	prf := &SampleFileShareProfiles[0]
+	var req = &pb.CreateFileShareOpts{
+		Id:          "d2975ebe-d82c-430f-b28e-f373746a71ca",
+		Name:        "sample-fileshare-01",
+		Description: "This is a sample fileshare for testing",
+		Size:        int64(1),
+		Profile:     prf.ToJson(),
+		Context:     c.NewAdminContext().ToJson(),
+	}
+	var fileshare = &SampleFileShares[0]
+	mockClient := new(dbtest.Client)
+	mockClient.On("GetFileShare", c.NewAdminContext(), req.Id).Return(&SampleFileShares[0], nil)
+	mockClient.On("GetDock", c.NewAdminContext(), "b7602e18-771e-11e7-8f38-dbd6d291f4e0").Return(&SampleDocks[0], nil)
+	mockClient.On("GetFileShareDefaultProfile", c.NewAdminContext()).Return(&SampleFileShareProfiles[0], nil)
+	mockClient.On("GetProfile", c.NewAdminContext(), "1106b972-66ef-11e7-b172-db03f3689c9c").Return(&SampleFileShareProfiles[0], nil)
+	mockClient.On("UpdateStatus", c.NewAdminContext(), fileshare, fileshare.Status).Return(nil)
+	db.C = mockClient
+
+	var ctrl = &Controller{
+		selector: &fakeSelector{
+			res: &model.StoragePoolSpec{
+				BaseModel: &model.BaseModel{
+					Id: "bdd44c8e-b8a9-488a-89c0-d1e5beb902dg",
+				},
+				DockId: "b7602e18-771e-11e7-8f38-dbd6d291f4e0",
+			},
+			err: nil,
+		},
+		fileshareController: NewFakeFileShareController(),
+	}
+	if _, err := ctrl.CreateFileShare(context.Background(), req); err != nil {
+		t.Errorf("failed to create fileshare, err is %v\n", err)
+	}
+
+}
+
+func TestDeleteFileShare(t *testing.T) {
```

Comment:
What about testFileShareAcl method?

Reply:
We have to first merge #948 till then i can't add fileshareacl test cases, it is failing with existing master branch code.

Reply:
Because existing code changes fileshare status after creating fileshare acl